WordPress/gutenberg
# Make warning message dynamic when deleting certain blocks like Query Loop block

fixes: #63350



## What?
Updated the static warning message shown when deleting a block to dynamically include the block and template name.

## Why?
The static message lacked clarity, leading to user confusion about which blocks or templates would be affected. Providing context in the message improves usability and reduces ambiguity.

## How?

1. Made the warning message dynamic to display:
- The name of the block being deleted.
- The template name where the block is being removed.
2. Adjusted the logic to fetch block and template names dynamically and integrate them into the alert.

## Testing Instructions

1. Add a Query Loop block to a template.
2. Delete the block and observe the warning message.
3. Confirm the message includes the block name and template name

## Screenshots or screencast 
| Singular message | Plural message |
|--|--|
| <img width="1470" alt="Screenshot 2024-11-26 at 1 10 20 AM" src="https://github.com/user-attachments/assets/c8da4103-f3c3-4276-8f10-630ff8741e47"> | <img width="1470" alt="Screenshot 2024-11-26 at 1 10 01 AM" src="https://github.com/user-attachments/assets/d1ffc216-31af-4c89-94fa-1458589ceee0"> |
